pkg/poll/poll.go

```diff
@@ -427,14 +427,29 @@ func (p *pollActions) forBuild(ctx context.Context, obj *unstructured.Unstructur
 		return errors.Wrap(err, "Could not get BuildList")
 	}
 
-	for _, build := range builds.Items {
-		callback := makeStatusCallback("Complete", "status", "phase")
-		if err := p.forResourceFullAvailability(ctx, &build, callback); err != nil {
+	var build *unstructured.Unstructured
+	for _, b := range builds.Items {
+		slice, found, err := unstructured.NestedSlice(b.Object, "metadata", "ownerReferences")
+		if err != nil {
 			return err
 		}
+		found = false
+		for _, element := range slice {
+			if name, ok := element.(map[string]interface{})["name"]; ok && name == obj.GetName() {
+				found = true
+				break
+			}
+		}
+		if found {
+			build = &b
+			break
+		}
```

Comment:
Can't this whole `if` statement be inside the `if` in line 438 before `break`?

Reply:
Changed this to a different approach without using a bool.